src/main/java/com/rackspacecloud/metrics/ingestionservice/listeners/rawlisteners/RawListener.java

```diff
@@ -44,50 +56,48 @@ public RawListener(InfluxDBHelper influxDBHelper, MeterRegistry registry) {
             containerFactory = "batchFactory",
             errorHandler = "listenerErrorHandler"
     )
-    public boolean listenUnifiedMetricsTopic(
-            @Payload final List<Metric> records,
+    public void listenUnifiedMetricsTopic(
+            @Payload final List<ExternalMetric> records,
             @Header(KafkaHeaders.RECEIVED_PARTITION_ID) final int partitionId,
             @Header(KafkaHeaders.OFFSET) final long offset,
             final Acknowledgment ack) throws Exception {
 
+        registry.gauge("ingestion.records.count", Arrays.asList(rawListenerTag), gaugeRecordsCount);
+        gaugeRecordsCount.set(records.size());
+
+        long batchProcessingStartTime = System.currentTimeMillis();
         batchProcessedCount++;
 
-        Map<String, List<String>> tenantPayloadsMap =
+        // Prepare the payloads to ingest
+        Map<TenantIdAndMeasurement, List<String>> tenantPayloadsMap =
                 RawMetricsProcessor.getTenantPayloadsMap(partitionId, offset, records);
 
-        boolean isInfluxDbIngestionSuccessful = writeIntoInfluxDb(tenantPayloadsMap);
+        writeIntoInfluxDb(tenantPayloadsMap);
 
-        return processPostInfluxDbIngestion(records.toString(),
-                partitionId, offset, ack, isInfluxDbIngestionSuccessful);
+        processPostInfluxDbIngestion(records.toString(), partitionId, offset, ack);
```

Comment:
Computing the `toString` of the records collection seems excessive when debug logs are not enabled. You could pass `records` directly and then let slf4j make the optimization here

https://github.com/racker/ceres-ingestion-service/blob/ccc7604d4a83f6829d3545c7586f766dd89e8b93/src/main/java/com/rackspacecloud/metrics/ingestionservice/listeners/UnifiedMetricsListener.java#L37